app/models/subscription.rb

```diff
@@ -51,11 +53,30 @@ def new_comment_by_someone_else?
   end
 
   def last_author
-    (discussion.comments.last || discussion).author
+    (last_comment || discussion).author
   end
 
-  def notify
-    NotificationsMailer.new_comment(id).deliver_later
+  def last_comment
+    discussion.comments.last
+  end
+
+  def notify_new_discussion
+    return unless new?
+    user.notifications.new_discussion(discussion)
```

Comment:
Yeah, that's probably a good idea. 

Either that or delegating through the user, but that would augment the user's api, which will tend to already be big, since it's a god class.

> On 22 Jan 2015, at 00:22, Miguel Palhas <notifications@github.com> wrote:
> 
> In app/models/subscription.rb:
> 
> >    end
> >  
> > -  def notify
> > -    NotificationsMailer.new_comment(id).deliver_later
> > +  def last_comment
> > +    discussion.comments.last
> > +  end
> > +
> > +  def notify_new_discussion
> > +    return unless new?
> > +    user.notifications.new_discussion(discussion)
> I'm not sure I understand that last comment
> 
> But anyway, I prefer to fix the LOD violation, actually. I'm just not sure how. Should I pass the user_id as an argument as well? (making them named arguments probably)
> 
> —
> Reply to this email directly or view it on GitHub.
>